sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -90,6 +94,10 @@ public RxGatewayStoreModel(
         this.sessionContainer = sessionContainer;
     }
 
+    void setCollectionCache(RxClientCollectionCache collectionCache) {
```

Comment:
can we pass in the constructor if possible?

Reply:
Same as this comment in other session token pr -- https://github.com/Azure/azure-sdk-for-java/pull/25555#discussion_r754658494 
To fill the cache we need GatewayStoreModel object ready itself, and master partition requests will work without any caches as we don't need them. That is the reason we cant pass cache to GatewayStoreModel constructor , and have to set them explicitly once we initialize them , which is after GatewayStoreModel object creation.

Reply:
doesn't this cause any race condition where one is ready but the other is not ready (set method not invoked yet)?

Reply:
Discussed offline , there will be no race condition , because this is getting set only one time inside client initialization method